grug/storage/src/codec.rs

```diff
@@ -5,10 +5,11 @@ use {
     },
     prost::Message,
     serde::{de::DeserializeOwned, ser::Serialize},
+    std::fmt::Debug,
 };
 
 /// A marker that designates encoding/decoding schemes.
-pub trait Codec<T> {
+pub trait Codec<T>: Debug + Clone + Copy + PartialEq + Eq {
```

Comment:
What's the purpose of this? I deleted it and everything still compiles and works...

Reply:
I figured since we wanted all Codecs to implement these we might as well require it so that we don't forget if we add new Codecs.